### ISSUE TITLE
Process.Status on Windows

### ISSUE THREAD

**Issue by bantmen** [shirou/gopsutil#1016]
**Is your feature request related to a problem? Please describe.**

Process.Status is not yet implemented for [Windows](https://github.com/shirou/gopsutil/blob/afe0c04c5d5a6cdf653565e4cbd2616657e3542c/process/process_windows.go#L341). Any thoughts on how we can go about implementing it?

**Additional context**

Python's psutil seems to have implemented it [here](https://github.com/giampaolo/psutil/blob/ecdaac33f5945be19090531be892791ac99fc597/psutil/arch/windows/services.c#L280).

**Comment by Lomanic** [comment]
The code you linked is for [windows services](https://psutil.readthedocs.io/en/latest/#psutil.WindowsService.status), not processes.

[The process.status() function is defined here](https://github.com/giampaolo/psutil/blob/ecdaac33f5945be19090531be892791ac99fc597/psutil/_pswindows.py#L1053-L1058), [it itself calls `psutil_proc_is_suspended`](https://github.com/giampaolo/psutil/blob/ecdaac33f5945be19090531be892791ac99fc597/psutil/_psutil_windows.c#L1163-L1184), [which itself calls `psutil_get_proc_info`](https://github.com/giampaolo/psutil/blob/ecdaac33f5945be19090531be892791ac99fc597/psutil/arch/windows/process_info.c#L585-L647) [which itself calls win32 `NtQuerySystemInformation` with `SystemProcessInformation`](https://docs.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntquerysysteminformation#systemprocessinformation), [which results](https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ex/sysinfo/process.htm) are hard to parse in Go.

It should be checked if there's another win32 API to retrieve threads state instead of NtQuerySystemInformation, we usually managed to use documented functions to get process info instead, but a quick search only return results with NtQuerySystemInformation.

https://stackoverflow.com/a/22949726 for some c++ code
